wallet/src/lib.rs

```diff
@@ -114,16 +95,18 @@ pub fn run(conf: Config) -> Result<(), Error> {
     // Initialize actors inside system context
     system.block_on(async {
         let node_subscriptions = Arc::new(Mutex::new(Default::default()));
-        let node_client_actor = JsonRpcClient::start_with_subscriptions(
-            node_client_url.as_ref(),
-            node_subscriptions.clone(),
-        )
-        .map_err(|_| app::Error::NodeNotConnected)?;
-        let node_client = Arc::new(NodeClient {
+        let (node_client_actor, url) =
+            JsonRpcClient::start_with_subscriptions(node_urls.clone(), node_subscriptions.clone())
+                .map_err(|_| app::Error::NodeNotConnected)?;
+        let node_client = Arc::new(app::NodeClient {
             actor: node_client_actor,
-            url: node_client_url.clone(),
+            url,
         });
 
+        // Trigger connection validation. Due to how the JSON-RPC client works, it will keep
+        // retrying connection using a different URL each time.
+        node_client.valid_connection().await;
```

Comment:
What happens if none of the URLs work? For example in case the user does not have an internet connection at the moment. Does he get a "Node disconnected" error message?

Reply:
Yes, he does, that remains unchanged.

Reply:
Cool, I was wondering if this await could cause some kind of infinite loop here and break everything.